Expensify/App
# Handle unauthenticated users when creating a new workspace

### Details

If an unauthenticated user navigates to https://expensify.cash/workspace/new, we redirect them to the signin page and, after signing in, send them back to https://expensify.cash/workspace/new.

cc @marcaaron 

### Fixed Issues

Fixes https://github.com/Expensify/Expensify/issues/166528

### Tests
1. Open e.cash and log out
2. Manually navigate to http://localhost:8080/workspace/new
    - Notice you get redirected back to the login page (http://localhost:8080/)
3. Sign in normally
4. Notice you get redirected to http://localhost:8080/workspace/new again, but since you're authenticated you see the actual new workspace page!

### QA Steps
1. Open e.cash and log out
2. Manually navigate to https://staging.expensify.cash/workspace/new
    - Notice you get redirected back to the login page (https://staging.expensify.cash/)
3. Sign in normally
4. Notice you get redirected to https://staging.expensify.cash/workspace/new again, but since you're authenticated you see the actual new workspace page!

### Tested On

- [ ] Desktop (can't navigate to https://expensify.cash/workspace/new as unauthenticated user)
- [ ] iOS (deep linking to https://expensify.cash/workspace/new not needed yet)
- [ ] Android (deep linking to https://expensify.cash/workspace/new not needed yet)

### Screenshots


#### Web


https://user-images.githubusercontent.com/3885503/121933619-9966fb80-cd46-11eb-92f1-a87516cd16e0.mov

#### Mobile Web


https://user-images.githubusercontent.com/3885503/121933841-df23c400-cd46-11eb-9913-efee28fbb0bf.mov



#### Desktop

N/A

#### iOS

N/A

#### Android

N/A
